### PR TITLE
feat(date-picker,input-date-picker): add support for built-in translation

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #4961

## Summary

This PR will add support for translations in `calcite-date-picker` and `calcite-input-date-picker` components.

### CODE REVIEW

File: src/components/color-picker/color-picker.e2e.ts [hunk 1]
```diff
@@ -319,7 +319,7 @@ describe("calcite-color-picker", () => {
   };
 
   function assertUnsupportedValueMessage(value: string | object | null, format: string): void {
-    expect(consoleSpy).toBeCalledTimes(1);
+    expect(consoleSpy).toHaveBeenCalledTimes(1);
```
Comment (jcfranco) [inline src/components/color-picker/color-picker.e2e.ts:322]: This is unrelated, please revert.

File: src/components/date-picker/date-picker.e2e.ts [hunk 2]
```diff
@@ -240,4 +227,6 @@ describe("calcite-date-picker", () => {
 
     expect(await date.getProperty("intlYear")).toEqual("Year");
   });
+
+  it("supports translations", () => t9n("<calcite-date-picker> </calcite-date-picker>"));
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.e2e.ts:231]: Our commonTest helpers take the tag name for easier setup:

```tsx
  it("supports translations", () => t9n("calcite-date-picker"));

```

File: src/components/date-picker/date-picker.tsx [hunk 3]
```diff
@@ -120,22 +128,25 @@ export class DatePicker implements LocalizedComponent {
    * Localized string for "previous month" (used for aria label)
    *
    * @default "Previous month"
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:130]: I think these no longer need defaults since it depends on the locale. I've been removing these entries in my PRs. @geospatialem @benelan WDYT?

File: src/components/date-picker/date-picker.tsx [hunk 4]
```diff
@@ -120,22 +128,25 @@ export class DatePicker implements LocalizedComponent {
...
+   * @deprecated - translations are now built-in, if you need to override a string, please use `messageOverrides`
    */
-  @Prop() intlPrevMonth?: string = TEXT.prevMonth;
+  @Prop() intlPrevMonth?: string;
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:133]: IMO, we don't need to mark props as optional as Stencil will do this for us for the generated custom element type.

File: src/components/date-picker/date-picker.tsx [hunk 5]
```diff
@@ -202,13 +234,16 @@ export class DatePicker implements LocalizedComponent {
 
   @State() globalAttributes = {};
 
+  @State() defaultMessages: Messages;
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:237]: This and other state props should be moved to the private state/prop section.

File: src/components/date-picker/date-picker.tsx [hunk 6]
```diff
@@ -202,13 +234,16 @@ export class DatePicker implements LocalizedComponent {
 
   @State() globalAttributes = {};
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:235]: Sidebar: this doesn't seem to be used at all.

File: src/components/date-picker/date-picker.tsx [hunk 7]
```diff
@@ -462,9 +504,10 @@ export class DatePicker implements LocalizedComponent {
         <calcite-date-picker-month-header
           activeDate={activeDate}
           headingLevel={this.headingLevel || HEADING_LEVEL}
-          intlNextMonth={this.intlNextMonth}
-          intlPrevMonth={this.intlPrevMonth}
-          intlYear={this.intlYear}
+          intlNextMonth={this.messages.nextMonth}
+          intlPrevMonth={this.messages.prevMonth}
+          intlYear={this.messages.year}
+          lang={this.effectiveLocale}
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:510]: We don't need to set lang here nor pass the locale, right?

File: src/components/date-picker/date-picker.tsx [hunk 8]
```diff
@@ -462,9 +504,10 @@ export class DatePicker implements LocalizedComponent {
         <calcite-date-picker-month-header
           activeDate={activeDate}
           headingLevel={this.headingLevel || HEADING_LEVEL}
-          intlNextMonth={this.intlNextMonth}
-          intlPrevMonth={this.intlPrevMonth}
-          intlYear={this.intlYear}
+          intlNextMonth={this.messages.nextMonth}
```
Comment (jcfranco) [inline src/components/date-picker/date-picker.tsx:507]: I think this is fine to keep (month-header not being a localized/t9n component) since it's a supporting component that won't be used outside of this context. Maybe we can add a note here?

File: src/components/input-date-picker/input-date-picker.tsx [hunk 9]
```diff
@@ -522,6 +529,7 @@ export class InputDatePicker
                   intlYear={this.intlYear}
                   max={this.max}
                   maxAsDate={this.maxAsDate}
+                  messageOverrides={this.messageOverrides}
```
Comment (jcfranco) [inline src/components/input-date-picker/input-date-picker.tsx:532]: Can you update the conventions doc and/or util doc about this scenario (non-localized/t9n component forwarding props)?

**Review by jcfranco — APPROVED** [review]
Aside from a few comments, this LGTM! 💥

File: src/components/date-picker/date-picker.tsx (hunk 7 above)
Comment (anveshmekala) [reply src/components/date-picker/date-picker.tsx:510]: Nope, we don't. It slipped through while merging locally. Good catch.

File: src/components/date-picker/date-picker.tsx (hunk 8 above)
Comment (anveshmekala) [reply src/components/date-picker/date-picker.tsx:507]: I think that's given considering it is not documented as a public component.

File: src/components/color-picker/color-picker.e2e.ts (hunk 1 above)
Comment (anveshmekala) [reply src/components/color-picker/color-picker.e2e.ts:322]: Its' the pre-commit linting job doing the change. Once #5431 lands-in, it will disappear.

File: src/components/date-picker/date-picker.tsx (hunk 6 above)
Comment (eriklharper) [reply src/components/date-picker/date-picker.tsx]: This is for accessing the global `lang` attribute.

File: src/components/date-picker/date-picker.tsx (hunk 4 above)
Comment (anveshmekala) [reply src/components/date-picker/date-picker.tsx:133]: marking it as optional helps us leverage documenting the prop as optional for the user.

File: src/components/date-picker/date-picker.tsx (hunk 6 above)
Comment (anveshmekala) [reply src/components/date-picker/date-picker.tsx]: that's wired-up in the new localization interface.

Comment (jcfranco) [reply src/components/date-picker/date-picker.tsx]: `LocalizedComponent` does not rely on this.

File: src/components/color-picker/color-picker.e2e.ts (hunk 1 above)
Comment (jcfranco) [reply src/components/color-picker/color-picker.e2e.ts:322]: Oh, I see. Ignore my comment then.

File: src/components/date-picker/date-picker.tsx (hunk 8 above)
Comment (jcfranco) [reply src/components/date-picker/date-picker.tsx:506]: We could make it clearer by adding a note under conventions/t9n module about internal components only needing to provide a prop to forward messages (`messages`?).
